harness/determined/common/api/_util.py

```diff
@@ -135,3 +135,101 @@ def _task_is_done_loading() -> Tuple[bool, Optional[str]]:
 
     err_msg = util.wait_for(_task_is_done_loading, timeout=300, interval=1)
     return err_msg
+
+
+def create_assignment_request(
+    session: api.Session,
+    role_name: str,
+    workspace_name: Optional[str] = None,
+    username_to_assign: Optional[str] = None,
+    group_name_to_assign: Optional[str] = None,
+) -> Tuple[List[bindings.v1UserRoleAssignment], List[bindings.v1GroupRoleAssignment]]:
+    if (username_to_assign is None) == (group_name_to_assign is None):
+        # XXX: very weird that we use api.errors.BadRequestException here!
+        raise api.errors.BadRequestException(
+            "must provide exactly one of --username-to-assign or --group-name-to-assign"
+        )
```

Comment:
wanna call this out:

I know we're using api.errors.BadRequestException to surface a nice user-facing error message.

I'm not happy with the way we're using it, but wanted to know if you had a particular idea as to how to raise a not-overly-CLI-specific error in a "helper function" that, let's be real, is mostly only for the CLI.  And a couple e2e tests.

Reply:
Ok, it sounds like since neither you nor I are sure what to do here, I'll just let it be.

I can explain why I don't like it in its current state, but I can't explain what would be better, and since at the end of the day the current form makes for a very nice user experience, changing it now feels like churn that would be unlikely to "stick" long-term.  Better to wait until we have a clearer motivating example as to what should be done, and fix according to that motivating example.

Reply:
Word. I support that approach.